tests/sentry/rules/conditions/test_event_attribute.py

```diff
@@ -370,6 +379,34 @@ def test_error_unhandled(self):
         )
         self.assertDoesNotPass(rule, event)
 
+    def test_error_main_thread(self):
+        event = self.get_event()
+        rule = self.get_rule(
+            data={"match": MatchType.EQUAL, "attribute": "error.main_thread", "value": "True"}
+        )
+        self.assertPasses(rule, event)
+
+        rule = self.get_rule(
+            data={"match": MatchType.EQUAL, "attribute": "error.main_thread", "value": "False"}
+        )
+        self.assertDoesNotPass(rule, event)
+
+    def test_error_main_thread_no_threads(self):
+        event = self.get_event(
+            threads={
+                "values": [],
+            }
+        )
+        rule = self.get_rule(
+            data={"match": MatchType.EQUAL, "attribute": "error.main_thread", "value": "True"}
+        )
+        self.assertDoesNotPass(rule, event)
+
+        rule = self.get_rule(
+            data={"match": MatchType.EQUAL, "attribute": "error.main_thread", "value": "False"}
+        )
+        self.assertDoesNotPass(rule, event)
```

Comment:
Those tests are failing, most likely I have to address https://github.com/getsentry/sentry/pull/46183/files#r1148928431 first